src/test/java/org/opensearch/knn/profiler/VectorProfilerTests.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.profiler;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.MockitoAnnotations;
+import org.opensearch.test.OpenSearchTestCase;
+
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class VectorProfilerTests extends OpenSearchTestCase {
```

Comment:
Does the profiler need to be singleton? This is one of the problems with going singleton route. It can complicate the testing. Instead, we can create via the KNNPLugin createComponents and pass into the codec/required transport actions from there.